### PR TITLE
Refactor LoaderCutsceneDAT and fix cutscenes on macOS

### LINKED ISSUE #378: Cutscenes not being rendered on MacOS
I get this on MacOS 10.13.2. I can't do anything until a cutscene ends. The sound is working. I can't skip a cutscene either.

screen shot 2018-01-29 at 23 40 49

### PR DESCRIPTION
**Pull request by emilisb**
Refactored Cutscene DAT Loader. Fixes #378 . The old code wasn't working on macOS.

screen shot 2018-06-08 at 19 31 34

### CODE REVIEW

File: rwengine/src/loaders/LoaderCutsceneDAT.cpp [hunk 1]
```diff
@@ -16,17 +18,21 @@ void LoaderCutsceneDAT::load(CutsceneTracks &tracks, const FileHandle& file) {
...
-        ss >> t;
-        ss.ignore(2, ',');
-        ss >> z;
-        ss.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+        std::string line;
+        ss >> line;
+
+        boost::split(lineParts, line, boost::is_any_of(","));
+
+        float t = stof(lineParts[0]);
```
Comment (husho) [inline rwengine/src/loaders/LoaderCutsceneDAT.cpp]: Better to use ```std::vector::at``` to check bounds, don't trust input data

Comment (emilisb) [reply rwengine/src/loaders/LoaderCutsceneDAT.cpp]: Good point. Fixed, thanks :)

File: rwengine/src/loaders/LoaderCutsceneDAT.cpp [hunk 2]
```diff
@@ -16,17 +18,21 @@ void LoaderCutsceneDAT::load(CutsceneTracks &tracks, const FileHandle& file) {
...
-        float t = 0.f;
-        float z = 0.f;
-        ss >> t;
-        ss.ignore(2, ',');
-        ss >> z;
-        ss.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+        std::string line;
+        ss >> line;
+
+        boost::split(lineParts, line, boost::is_any_of(","));
```
Comment (christoph-heiss) [inline rwengine/src/loaders/LoaderCutsceneDAT.cpp]: Wasn't there once some endeavor (#317) for getting rid of the boost dependency, at least for anything except the testcases? 

(IMHO, I do not really see the advantage here over using e.g. `std::getline` with `,` as separator, or just fixing the existing code.)

**Comment by darkf** [comment]
I agree that boost should be avoided where possible

**Comment by JayFoxRox** [comment]
@emilisb it has been almost 2 weeks since you've sent this PR and more than 1 week since the first feedback:

- What's your plan with this PR?
    - Are you still around?
    - Are you going to address the feedback?
- Also, is there an issue for this problem (if so, please mention it in the PR description)?
    - What is the actual problem with the current code?

**Comment by JayFoxRox** [comment]
Closed.

Continued in #533 which does the same boost-less, has better documentation and active code author.
Thanks to @emilisb for pointing out a possible solution and motivating the work :)

**Pull request closed without merging**